pkg/controller/healthcheck/healtcheck_actuator.go

```diff
@@ -93,28 +96,74 @@ type checkResultForConditionType struct {
 func (a *Actuator) ExecuteHealthCheckFunctions(ctx context.Context, request types.NamespacedName) (*[]Result, error) {
 	_, shootClient, err := util.NewClientForShoot(ctx, a.seedClient, request.Namespace, client.Options{})
 	if err != nil {
-		msg := fmt.Sprintf("failed to create shoot client in namespace '%s'", request.Namespace)
-		a.logger.Error(err, msg)
-		return nil, fmt.Errorf(msg)
+		msg := fmt.Errorf("failed to create shoot client in namespace '%s'", request.Namespace)
+		a.logger.Error(err, msg.Error())
+		return nil, msg
 	}
-	channel := make(chan channelResult)
-	var wg sync.WaitGroup
-	wg.Add(len(a.healthCheckMappings))
-	for healthCheck, healthConditionType := range a.healthCheckMappings {
+
+	var (
+		channel = make(chan channelResult)
+		wg      sync.WaitGroup
+	)
+
+	wg.Add(len(a.healthChecks))
+	for _, hc := range a.healthChecks {
 		// clone to avoid problems during parallel execution
-		check := healthCheck.DeepCopy()
+		check := hc.HealthCheck.DeepCopy()
 		check.InjectSeedClient(a.seedClient)
 		check.InjectShootClient(shootClient)
 		check.SetLoggerSuffix(a.provider, a.extensionKind)
-		go func(ctx context.Context, request types.NamespacedName, check HealthCheck, healthConditionType string) {
+
+		go func(ctx context.Context, request types.NamespacedName, check HealthCheck, preCheckFunc PreCheckFunc, healthConditionType string) {
 			defer wg.Done()
+
+			obj := a.getExtensionObjFunc()
+			if err := a.seedClient.Get(ctx, client.ObjectKey{Namespace: request.Namespace, Name: request.Name}, obj); err != nil {
+				channel <- channelResult{
+					healthCheckResult: &SingleCheckResult{
+						IsHealthy: false,
+						Detail:    err.Error(),
+						Reason:    "ReadExtensionObjectFailed",
+					},
+					error:               err,
+					healthConditionType: healthConditionType,
+				}
+				return
+			}
+
+			cluster, err := extensionscontroller.GetCluster(ctx, a.seedClient, request.Namespace)
```

Comment:
I would suggest to move the get for cluster and extension object only if the `preCheckFunc != nil`. Currently the get calls  will be executed even if the `preCheckFunc  == nil`.

Reply:
done, good idea :)

Reply:
that was fast